### PR TITLE
Add ManaToki

### PR DESCRIPTION
**Pull request by nar1n**

### CODE REVIEW

File: src/ManaToki/ManaToki.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        const response = await this.requestManager.schedule(request, 3)
+        const $ = this.cheerio.load(response.data)
+        const manga = parseSearch($)
+        
+        return createPagedResults({
+            results: manga,
+        })
+    }
+
+    async filterUpdatedManga(mangaUpdatesFoundCallback: (updates: MangaUpdates) => void, time: Date, ids: string[]): Promise<void> {
```
Comment (AdrienSeon) [inline src/ManaToki/ManaToki.ts:200]: I got [IP banned](https://discord.com/channels/673606787290759230/705193595715452958/832767531454169108) from 2 different IPs while testing the method, I don't know if we should really implement it or create a really slow `requestManager` maybe.
I'll do more tests if when I'm unbanned if I ever am...

File: src/tests/ManaToki.test.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        expect(result.title, "No title").to.be.not.null;
+    });
+
+    it("Testing Home-Page aquisition", async() => {
+        let homePages = await wrapper.getHomePageSections(source)
+        expect(homePages, "No response from server").to.exist
+        expect(homePages[0], "No updates section available").to.exist
+        expect(homePages[1], "No list section available").to.exist
+    })
+
```
Comment (AdrienSeon) [inline src/tests/ManaToki.test.ts:80]: Could you add these tests for the next time we're testing this source ?

```js
it("Testing home page results for 최신화", async () => {
    const resultsPage1 = await wrapper.getViewMoreItems(source, "list", { page: 1 }, 1);
    const resultsPage3 = await wrapper.getViewMoreItems(source, "list", { page: 3 }, 1);

    expect(resultsPage1, "No results for page 1 for this section").to.exist;
    expect(resultsPage3, "No results for page 3 for this section").to.exist;
    expect(resultsPage1, "Page 1 and 3 are the same").to.not.be.eql(resultsPage3);

    const data = resultsPage1![0];

    expect(data.id, "No ID present").to.exist;
    expect(data.image, "No image present").to.exist;
    expect(data.title.text, "No title present").to.exist;
});

it("Testing home page results for 만화목록", async () => {
    const resultsPage1 = await wrapper.getViewMoreItems(source, "list", { page: 1 }, 1);
    const resultsPage3 = await wrapper.getViewMoreItems(source, "list", { page: 3 }, 1);

    expect(resultsPage1, "No results for page 1 for this section").to.exist;
    expect(resultsPage3, "No results for page 3 for this section").to.exist;
    expect(resultsPage1, "Page 1 and 3 are the same").to.not.be.eql(resultsPage3);

    const data = resultsPage1![0];

    expect(data.id, "No ID present").to.exist;
    expect(data.image, "No image present").to.exist;
    expect(data.title.text, "No title present").to.exist;
});

it("Testing Notifications", async () => {
    const updates = await wrapper.filterUpdatedManga(source, new Date("2021-03-26"), [mangaId]);

    expect(updates, "No server response").to.exist;
    expect(updates, "Empty server response").to.not.be.empty;
    expect(updates[0].ids, "No updates").to.not.be.empty;
});
```

**Review by AdrienSeon — CHANGES_REQUESTED** [review]
Source looking really good fully tested and working great, just have to talk about the `filterUpdatedManga` method (cf. my comment in the review).

@ConradWeiser Could we have your opinion on the matter?

File: src/tests/ManaToki.test.ts (hunk 2 above)
Comment (nar1n) [reply src/tests/ManaToki.test.ts:112]: Added, thank you.

File: src/ManaToki/ManaToki.ts (hunk 1 above)
Comment (ConradWeiser) [reply src/ManaToki/ManaToki.ts:200]: The first test I ran also IP banned me. I switched to a VPN and tried changing the requests per second from `2` to `1`, and still got IP banned with this given test case.

I think this method should be commented out.

**Review by ConradWeiser — CHANGES_REQUESTED** [review]
As stated in a discussion above, the update function is verified to cause IP bans. 
Comment that out for now, and I think this is good to go!

File: src/ManaToki/ManaToki.ts [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+export const ManaTokiInfo: SourceInfo = {
+    version: '1.0.3',
+    name: '마나토끼',
+    icon: 'icon.png',
+    author: 'nar1n',
+    authorWebsite: 'https://github.com/nar1n',
+    description: 'Extension that pulls manga from ManaToki',
+    hentaiSource: false,
+    websiteBaseURL: MANATOKI_DOMAIN
+}
```
Comment (AdrienSeon) [inline src/ManaToki/ManaToki.ts:28]: Last thing that I didn't catch earlier, sorry, could you add the source language as a tag 
```js
sourceTags: [
        {
            text: "Korean",
            type: TagType.GREY
        }
    ]
```
Other that that good to go for me!

**Review by AdrienSeon — CHANGES_REQUESTED** [review]

**Review by Aur0raN — COMMENTED** [review]
Tested locally and I see no issues so far, Search and Discover works fine. Scrolled through a few manga as well.

**Review by AdrienSeon — APPROVED** [review]

**Review by ConradWeiser — APPROVED** [review]
Taking another pass on a local device, this checks out and works great! Thank you for the attention!

**Pull request merged**